indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -520,7 +530,7 @@ protected void reduce(
         int runningTotalLineCount = 0;
         long startTime = System.currentTimeMillis();
 
-        Set<String> allDimensionNames = Sets.newHashSet();
+        Set<String> allDimensionNames = Sets.newLinkedHashSet();
```

Comment:
This set gets populated by `inputRow.getDimensions()` which depends on the order in which rows appear. Are we sue the order gets preserved here? If I ask for `[a,b,c]` in my indexing spec, and if my first row only has dimension `c`, my second row only has dimension `b`, and my third row only has dimension `a`, then this set may end up with `[c,b,a]` instead of `[a,b,c]`